openwisp_radius/tests/test_api.py

```diff
@@ -2414,3 +2414,64 @@ def test_change_phone_number_restriction(self):
             self.assertEqual(user.phone_number, phone_number)
 
         app_settings.ALLOWED_MOBILE_PREFIXES = []
+
+
+class TestApiRejectPhoneToken(ApiTokenMixin, BaseTestCase):
+
+    _test_email = 'test@openwisp.org'
+
+    def setUp(self):
+        super().setUp()
+        radius_settings = self.default_org.radius_settings
+        radius_settings.sms_verification = False
+        radius_settings.sms_sender = '+595972157632'
+        radius_settings.sms_meta_data = {
+            'clientId': 3,
+            'clientSecret': 'p8pVUoqz7T4ArPEu1rcgzlsYJmnGCkkWHs2WAqqm',
+        }
+        radius_settings.full_clean()
+        radius_settings.save()
+
+    def test_register_201(self):
+        self._superuser_login()
+        self.assertEqual(User.objects.count(), 1)
+        url = reverse('radius:rest_register', args=[self.default_org.slug])
+        phone_number = '+393664255801'
+        r = self.client.post(
+            url,
+            {
+                'username': self._test_email,
+                'email': self._test_email,
+                'password1': 'password',
+                'password2': 'password',
+                'phone_number': phone_number,
+            },
+        )
+        self.assertEqual(r.status_code, 201)
+        self.assertIn('key', r.data)
+        self.assertEqual(User.objects.count(), 2)
+        user = User.objects.get(email=self._test_email)
+        self.assertTrue(user.is_member(self.default_org))
+        self.assertEqual(user.phone_number, phone_number)
```

Comment:
if SMS verification is disabled, I thought the phone number field should not be saved, but it looks like it does.
This is an issue, because mobile phone verification may be enabled later on by an org and they'll have active users with the phone number already filled, and may think it's been verified but it hasn't.

Could you help me to fix this in the API? It could be handled here:
https://github.com/openwisp/openwisp-radius/blob/master/openwisp_radius/api/serializers.py#L342-L352
If sms verification is not enabled, we could return an empty string, which should force the phone number to be empty.
The expected value in this case is an empty string.
If you can do this change, please do it in a separate commit (eg: `[change] Force User.phone_number to be empty if SMS verification disabled`).